### PR TITLE
DJ October 2021

### PR DESCRIPTION
**Pull request by xaur**
**DRAFT**: https://github.com/xaur/decred-news/blob/draft202110/journal/202110.md

**Dependencies**: Send notice in advance so that people can plan their work.

- [x] [title image](https://github.com/decred/dcrdesign/issues) _(notify by: 20th, ready by: 3rd)_
- [x] first time contributors
- [x] PR update _(notify by: 28th, ready by: 3rd)_

**Monthly snapshots**: Snapshot data that is hard to obtain later. Ideally, capture it at 1st day of the month at 00:00 UTC.

- [x] social media stats
  - accs reported in People (the "key" list)
  - all accs in [social-media-stats](https://github.com/decredcommunity/social-media-stats)
  - untracked candidate accounts (the "maybe" list)
  - see if any new accounts need to be tracked
- [x] Medium and Publish0x post stats
- [x] Reddit traffic stats
- [x] mining
  - pool hashrate absolute Ph/s and share % values
  - [miningpoolstats.stream](https://miningpoolstats.stream/decred)
  - [dcrstats.com/pow](https://dcrstats.com/pow)
  - luxor.tech API
  - save raw data where possible (JSON etc)
- [x] staking
  - ticket price [30-day average](https://dcrstats.com)
- [x] network stats
  - nodes at [Decred Mapper](https://nodes.jholdstock.uk/user_agents)
  - nodes at [dcrextdata](https://dcrextdata.planetdecred.org/nodes)
  - update the list of known [nodes and groups](https://github.com/decredcommunity/network-stats/tree/master/nodes)
- [x] vspd and dcrstakepool instance stats
- [x] [LN stats](https://ln-map.jholdstock.uk/)

**Writing**

- [x] scan GitHub repos for dev updates
- [x] complete Development _(by: 2nd)_
- [x] complete People _(by: 2nd)_
- [x] PO review Development and People
- [x] ask in #dev to review Development and People
- [x] complete Events
- [x] ask #events to review Events
- [x] complete Outreach
- [x] complete Media
- [x] ask #marketing to review Outreach+Events+Media
- [x] complete Governance
- [x] complete Network
- [x] complete Ecosystem
- [x] find interesting Reddit and Twitter threads
- [x] complete Discussions
- [x] complete Markets
- [x] complete Relevant External
- [x] ask quoted people to review their texts
- [x] complete Intro
- [x] PO review of all sections
- [x] ask #writers to review _(by: 3rd)_
- [x] incorporate all feedback
- [x] resolve all [TODOs](https://github.com/xaur/decred-news/blob/docs/guidelines.md#todos)
- [x] ensure that all people who contributed or gave feedback are in [credits](https://github.com/xaur/decred-news/blob/docs/guidelines.md#how-to-give-credit)
- [x] add [title image](https://github.com/xaur/decred-news/blob/docs/guidelines.md#title-image)
- [x] add other images
- [x] pre-release review: check spelling, grammar, test that links work
- [x] [lint](https://github.com/xaur/decred-news/blob/docs/guidelines.md#linting)
- [x] remove `{DRAFT}` from title

**Publishing**

- [x] publish on GitHub with a single commit to `gh-pages`
- [x] add all images to the [files repo](https://github.com/xaur/decred-journal-files)
- [x] publish on Medium
  - upload images
  - make Table of Contents
  - add to `/decred/journals`
  - add relevant tags
  - verify text previews shown at `/decred` and `/decred/journals`
  - check for formatting issues
- [x] publish on Publish0x
  - upload images
  - add relevant tags
  - check for formatting issues
- [x] post on r/decred
- [x] tweet via decredproject
  - add correct title image for Twitter
- [x] announce on: Matrix #dcr, t.me/DCRann, Blockfolio, CoinGecko, LinkedIn
- [x] announce on Instagram
- [x] fast-forward `master` to `gh-pages`
- [x] if changes are needed after the release, add commits to `gh-pages` (test in `draftXX` if needed) and update other places
- [x] [update index](https://github.com/xaur/decred-news/blob/docs/guidelines.md#updating-index)
- [x] add new issue to decred.org Press page ([coverage.yml](https://github.com/decred/dcrweb/blob/master/src/data/press/coverage.yml))
  - verify deployment

**Housekeeping**

- [x] archive draft history
  - create `tmp` at `master`, merge both `drafts` and the latest draft commit, fast-forward `drafts` to `tmp`
- [x] update `journal-template.md` in the `docs` branch
- [x] update `release-checklist-template.md` in the `docs` branch
- [x] start new cycle
- [x] delete the draft branch (will auto-close the PR)

### CODE REVIEW

File: journal/202110.md [hunk 1]
```diff
@@ -0,0 +1,450 @@
...
+**[dcrd](https://github.com/decred/dcrd)**
+
+_dcrd is a full node implementation that powers Decred's peer-to-peer network around the world._
+
+Merged changes:
+
+- updating of indexes (transaction, address, exists address) has been made [asynchronous](https://github.com/decred/dcrd/pull/2219) to speed up the critical path of block validation and connection code { any other big picture or fun stuff to mention for it? and does it help multipeer downloads? }
+- simplified to use only one latest [checkpoint](https://github.com/decred/dcrd/pull/2763), since intermediate checkpoints are no longer necessary with headers-first syncing { any impact on end users to mention, e.g. any perf gains? also, is [2555](https://github.com/decred/dcrd/pull/2555) the best link for "headers-first" sync in dcrd? }
+- improved consistency and clarity of `txscript` package [tests](https://github.com/decred/dcrd/pull/2757)
+- fixed handling of automatic ticket revocations when their block is [disconnected](https://github.com/decred/dcrd/pull/2768) { is this for reorg handling? }
```
Comment (davecgh) [inline journal/202110.md:44]: > { is this for reorg handling? }

Yes.

File: journal/202110.md [hunk 2]
```diff
@@ -0,0 +1,450 @@
...
+
+<a id="dcrd" />
+
+**[dcrd](https://github.com/decred/dcrd)**
+
+_dcrd is a full node implementation that powers Decred's peer-to-peer network around the world._
+
+Merged changes:
+
+- updating of indexes (transaction, address, exists address) has been made [asynchronous](https://github.com/decred/dcrd/pull/2219) to speed up the critical path of block validation and connection code { any other big picture or fun stuff to mention for it? and does it help multipeer downloads? }
```
Comment (davecgh) [inline journal/202110.md:41]: It allows for faster vote propagation and helps pave the way for other optimizations, sync models, and ultimately nicer data corruption recovery mechanisms.

File: journal/202110.md [hunk 3]
```diff
@@ -0,0 +1,450 @@
...
+- simplified to use only one latest [checkpoint](https://github.com/decred/dcrd/pull/2763), since intermediate checkpoints are no longer necessary with headers-first syncing { any impact on end users to mention, e.g. any perf gains? also, is [2555](https://github.com/decred/dcrd/pull/2555) the best link for "headers-first" sync in dcrd? }
```
Comment (davecgh) [inline journal/202110.md:42]: { any impact on end users to mention, e.g. any perf gains? also, is [2555](https://github.com/decred/dcrd/pull/2555) the best link for "headers-first" sync in dcrd? }

That link is correct.  This change won't have any noticeable impact on users, however, the aforementioned PR (2555) that makes multiple checkpoints no longer needed most definitely does.

File: journal/202110.md [hunk 4]
```diff
@@ -0,0 +1,450 @@
...
+
+**[Politeia](https://github.com/decred/politeia)**
+
+_Politeia is Decred's proposal system. It is used to request funding from the Decred treasury._
+
+User-facing changes:
+
+- added standard way to show any status change reason (not only censored but now also abandoned and closed)
+- support [multiple](https://github.com/decred/politeiagui/pull/2629) billing status changes. Ideally there should be only one (from Active to Completed or Closed), but this allows to correct admin mistakes. It is also possible to temporarily [disable](https://github.com/decred/politeia/pull/1533) status changes.
+- improved [identity error](https://github.com/decred/politeiagui/pull/2623) messages
```
Comment (lukebp) [inline journal/202110.md:93]: This improves and standardizes the identity error.  The same error is now displayed anytime the user attempts to write data to politeia without their active identity being loaded in the browser.  The error message directs them to their user details page to fix the issue and provides a link to do so.

File: journal/202110.md [hunk 5]
```diff
@@ -0,0 +1,450 @@
...
+
+<a id="politeia" />
+
+**[Politeia](https://github.com/decred/politeia)**
+
+_Politeia is Decred's proposal system. It is used to request funding from the Decred treasury._
+
+User-facing changes:
+
+- added standard way to show any status change reason (not only censored but now also abandoned and closed)
```
Comment (lukebp) [inline journal/202110.md:91]: The username of the admin the censored or abandoned the proposal is displayed along with the reason they gave for the status change.

Example: https://github.com/decred/politeiagui/pull/2625#issuecomment-965883410

File: journal/202110.md [hunk 6]
```diff
@@ -0,0 +1,450 @@
...
+- support [multiple](https://github.com/decred/politeiagui/pull/2629) billing status changes. Ideally there should be only one (from Active to Completed or Closed), but this allows to correct admin mistakes. It is also possible to temporarily [disable](https://github.com/decred/politeia/pull/1533) status changes.
```
Comment (lukebp) [inline journal/202110.md:92]: > Ideally there should be only one

Maybe specify that this defaults to only allowing a single billing status change, but that this is a configurable setting.  If an admin makes a mistake, the sysadmin can temporarily update the setting to allow for the mistake to be corrected.

File: journal/202110.md [hunk 7]
```diff
@@ -0,0 +1,450 @@
...
+- improved [identity error](https://github.com/decred/politeiagui/pull/2623) messages
+- ~14 bug fixes
```
Comment (lukebp) [inline journal/202110.md:94]: [2638](https://github.com/decred/politeiagui/pull/2638) is also a notable PR.  It fixes a handful of comment navigation UX issues that occured when navigating back and forth between the full comments section and a single comment thread.  Issues that were fixed:
- Comment filtering preferences (top, new, old, flat mode) were being lost in between thread navigation.
- The "Go Back" button was not working properly.
- Loading indiviudal threads was slow due to unnecessary requests being sent.
- UI flickering was occuring during navigation.

File: journal/202110.md [hunk 8]
```diff
@@ -0,0 +1,450 @@
...
+
+- politeiawww codebase reorganized to move legacy APIs into a [`legacy`](https://github.com/decred/politeia/pull/1523) package. This will make it easier to rewrite the user layer to use a plugin architecture and to allow for horizontal scaling.
+- extracted [`logger`](https://github.com/decred/politeia/pull/1527) package to allow plugins configure their logging and become self-contained
+- extracted [`websockets`](https://github.com/decred/politeia/pull/1529) package (will make it easier to scale the servers)
+- extracted [config](https://github.com/decred/politeia/pull/1536) handling into its own package and separated out settings for legacy API for easier removal in the future
+- moved [identity](https://github.com/decred/politeia/pull/1530) handling methods to more appropriate places
+- added a generic [session store](https://github.com/decred/politeia/pull/1555) (will replace the [legacy](https://github.com/decred/politeia/pull/1554) one that did not have proper separation of concerns)
+
+{ image candidates: maybe [2625](https://github.com/decred/politeiagui/pull/2625), or just skip this time }
```
Comment (lukebp) [inline journal/202110.md:120]: Up to you.

File: journal/202110.md (hunk 2 above)
Comment (xaur) [reply journal/202110.md]: Nice explainer! Added.

File: journal/202110.md (hunk 4 above)
Comment (xaur) [reply journal/202110.md]: Good detail. Added a slightly compressed version.

A side effect of such mentions is it helps to raise [awareness](https://github.com/decred/dcrdesign/issues/261) about user identity.

File: journal/202110.md (hunk 5 above)
Comment (xaur) [reply journal/202110.md]: Nice, added.

File: journal/202110.md (hunk 6 above)
Comment (xaur) [reply journal/202110.md]: Good details, added.

File: journal/202110.md (hunk 7 above)
Comment (xaur) [reply journal/202110.md]: Thanks for the clarification. I extracted comment navigation fixes into their own bullet with some compression (Pi section is getting beefy, which is great).

**Comment by xaur** [comment]
Hmm I pushed 2 commits but the second one c07e8571 is not show. Let's see if this comment pokes GitHub to flush the cache or something.

**Pull request closed without merging**